### PR TITLE
Split `check_style_impl` into `run_rubocop` and `run_shellcheck`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Also add JSON output for ShellCheck so we can output violations as annotations on GitHub Actions in the future.

### CODE REVIEW

File: Library/Homebrew/style.rb [hunk 1]
```diff
@@ -79,63 +115,98 @@ def check_style_impl(files, output_type,
...
+
+      case output_type
+      when :print
+        system shellcheck, "--format=tty", *args
+        $CHILD_STATUS.success?
+      when :json
+        result = system_command shellcheck, args: ["--format=json1", *args]
+        json = json_result!(result)
+
+        # Convert to same format as RuboCop offenses.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/style.rb:161]: Why?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @reitermarkus!

File: Library/Homebrew/style.rb (hunk 1 above)
Comment (reitermarkus) [reply Library/Homebrew/style.rb:161]: I think it's easier to return a unified result for both RuboCop and ShellCheck offenses this way.

Comment (MikeMcQuaid) [reply Library/Homebrew/style.rb:161]: Ah, unified result, makes sense 👍🏻

**Pull request merged**